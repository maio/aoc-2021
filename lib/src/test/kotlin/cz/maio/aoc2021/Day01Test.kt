package cz.maio.aoc2021

import cz.maio.ResourcesUtils.resourceLines
import org.approvaltests.Approvals
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day01Test {
    private val lines = resourceLines("/input-01.txt").map { it.toInt() }

    @Test
    fun `part 1`() {
        var previous: Int? = null
        var count = 0

        lines.forEach { current ->
            if (current > (previous ?: current)) {
                count++
            }
            previous = current
        }

        assertThat(count).isEqualTo(1581)
    }

    @Test
    fun `part 2`() {
        val windows = lines.windowed(size = 3, step = 1)

        val windowsToCompare = windows.zipWithNext()

        val count = windowsToCompare.count { (previous, current) ->
            current.sum() > previous.sum()
        }

        Approvals.verify(
            """
                first 4 lines = ${lines.take(4)}
                ...
                last 4 lines = ${lines.takeLast(4)}
                
                first windows to compare: ${windowsToCompare.first()}
                ...
                last windows to compare: ${windowsToCompare.last()}
                
                result = $count
            """.trimIndent()
        )
    }
}
