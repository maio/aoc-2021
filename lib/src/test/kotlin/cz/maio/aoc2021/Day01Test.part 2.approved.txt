first 4 lines = [176, 184, 188, 142]
...
last 4 lines = [8306, 8317, 8320, 8341]

first windows to compare: ([176, 184, 188], [184, 188, 142])
...
last windows to compare: ([8306, 8317, 8320], [8317, 8320, 8341])

result = 1618